### PR TITLE
qbe: 1.1 -> 1.2

### PR DESCRIPTION
**Pull request by onemoresuza**
## Description of changes

[Changelog](https://c9x.me/compile/releases.html).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fgaz** [comment]
@GrahamcOfBorg build hare

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 289276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qbe</li>
  </ul>
</details>

**Comment by onemoresuza** [comment]
> @GrahamcOfBorg build hare

`harec` and `hare` are using a pinned version of `qbe`. Locally, I've managed to build them with `qbe-1.2`.

After this PR is merged, both of those packages will be able to use the `qbe` package without the need of overriding it.

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 289276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qbe</li>
  </ul>
</details>

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 289276` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qbe</li>
  </ul>
</details>

**Comment by onemoresuza** [comment]
@fgaz, any idea why the tests are failing on darwin?

**Comment by fgaz** [comment]
> `cannot open '/tmp/qbe.zzzz.s'`

I guess it hardcodes /tmp when it should use $TMPDIR instead.

**Comment by onemoresuza** [comment]
I've submitted a [patch](https://lists.sr.ht/~mpu/qbe/patches/49613) upstream. Do you think it is better to wait for them or merge this PR and then create another one with the patch applied?

File: pkgs/development/compilers/qbe/default.nix [hunk 1]
```diff
@@ -1,23 +1,24 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchzip
 , callPackage
+,
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/qbe/default.nix:5]:
```suggestion
```

**Comment by fgaz** [comment]
@onemoresuza yes please apply the darwin patch in another pr

**Pull request merged**